app/code/community/Bolt/Boltpay/controllers/ApiController.php

```diff
@@ -195,10 +195,34 @@ public function hookAction()
                     ->setHeader("Retry-After", "86400")
                     ->setBody(json_encode(array('status' => 'failure', 'error' => array('code' => 6009, 'message' => $boltHelperBase->__('The order is on-hold and requires manual update before this hook is accepted') ))));
             } else {
-                // An invalid transition is treated as a late queue event and hence will be ignored
-                //Mage::log($errorMessage, null, 'bolt.log');
-                //Mage::log("Late queue event. Returning as OK", null, 'bolt.log');
-                $this->getResponse()->setHttpResponseCode(200);
+                /////////////////////////////////////////////////////////////
+                // An invalid transition is considered manually handled if
+                //      1.) It is not a credit hook
+                //      2.) It is moving from and to the same status
+                //      3.) The order is canceled, and therefore, nothing left to do, or
+                //      4.) The previous status is completed, and therefore, there is nothing left to do.
+                /////////////////////////////////////////////////////////////
+                if ($newTransactionStatus !== 'credit'
```

Comment:
we'll support multiple captures. I'm not sure what is the magento status after payment is partially captured (processing?) but we need to support that use case as well.